txnkv/transaction/txn.go

```diff
@@ -763,6 +763,9 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 			checkedExistence = true
 		}
 	}
+	if assignedPrimaryKey && lockCtx.LockOnlyIfExists && len(keys) == 1 {
```

Comment:
Here I prefer to make the `&& len(keys) == 1` part an assertion. For example, add
```go
if len(keys) != 1 {
	panic("LockOnlyIfExists only assigns the primary key when locking only one key")
}
```
Then, if `LockOnlyIfExists` extends its usage in the future, it's more possible for us to find we need to change here too.

Reply:
good idea